### DISCUSSION TITLE
Architecture change

### DISCUSSION THREAD

**Discussion by noraj** [mBouamama/rawsec_cli#202] [General]
I made a major architecture change on the data structure to be able to have Resources and Operating Systems are also available from the search engine!

https://gitlab.com/rawsec/rawsec-cybersecurity-list/-/commit/c9f762fb33cd535a2398db367d9843ff0463e96a

For operating_systems:

- `os` attribute -> `name`
- `link` attribute -> `website`

For resources:

- The nested links have been flattened
  - `links` attribute was removed
  - `website` and/or `source` attribute have been created

Eg.

```diff
-"links": [
-        {
-          "website": "https://bugbounty.jp/"
-        }
-      ],
+      "website": "https://bugbounty.jp/",
```

**Comment by tyki6** [comment]
hi @noraj ,
Sorry i didn"t check my github.
I create a pr for [fix new architecture](#203), and i release a new version 1.2.0 with this fix!

**Reply by noraj** [reply]
Awesome!

**Comment by noraj** [comment]
Hi @tyki6,

there are new architecture changes but only additions:

- a long time ago, I introduced a `keywords` attribute in the JSON, not displayed in the tables but mainly used for the search engine, optionally it could be displayed on rawsec-cli search results too
- not so long ago, I introduced a new meta category: certifications (see https://inventory.raw.pm/certifications.html) with 3 sub categories for now: Technical Defensive, Technical Offensive, Governance
